### PR TITLE
Update spelling in s2s authorization

### PR DESCRIPTION
**Pull request by saiprakash2**
Fixes https://github.com/matrix-org/matrix-spec/issues/1231

### CODE REVIEW

**Review by turt2live — CHANGES_REQUESTED** [review]
Thanks for the contribution! I've already fixed the PR title and description, but please review https://github.com/matrix-org/matrix-spec/blob/main/CONTRIBUTING.rst when you get a chance.

Primarily, this needs a changelog entry to cover the typo.

**Comment by saiprakash2** [comment]
I added the change log entry.

File: changelogs/server_server/newsfragments/1431.clarification [hunk 1]
```diff
@@ -0,0 +1 @@
+Update spelling in s2s authorization.
```
Comment (turt2live) [inline changelogs/server_server/newsfragments/1431.clarification:1]:
```suggestion
Fix various typos throughout the specification.
```

**Review by turt2live — APPROVED** [review]
Thanks!

**Pull request merged**